crates/matrix-sdk/src/widget_api/handler/capabilities.rs

```diff
@@ -17,48 +14,38 @@ use crate::widget_api::{
 #[allow(missing_debug_implementations)]
 #[derive(Default)]
 pub struct Capabilities {
-    // Room and state events use the same sender, reader, listener
-    // on the rust-sdk side room and state events dont make a difference for the transport.
-    // It is the widgets responsibility to differenciate and react to them accordingly.
-    pub event_listener: Option<UnboundedReceiver<MatrixEvent>>,
-
-    pub event_reader: Option<Box<dyn EventReader>>,
-    pub event_sender: Option<Box<dyn EventSender>>,
-    // TODO implement to_device_sender (not required for EC->EX)
-    // pub to_device_sender: Option<Box<dyn ToDeviceSender>>,
+    pub listener: Option<UnboundedReceiver<MatrixEvent>>,
+    pub reader: Option<Box<dyn EventReader>>,
+    pub sender: Option<Box<dyn EventSender>>,
 }
 
 #[async_trait]
-pub trait EventReader {
+pub trait EventReader: EventHandler + Send {
     async fn read(&self, req: ReadEventRequest) -> Result<ReadEventResponse>;
-    fn get_filter(&self) -> &Vec<Filter>;
 }
 
 #[async_trait]
-pub trait EventSender {
+pub trait EventSender: EventHandler + Send {
     async fn send(&self, req: SendEventRequest) -> Result<SendEventResponse>;
-    fn get_filter(&self) -> &Vec<Filter>;
 }
 
-#[async_trait]
-pub trait ToDeviceSender {
-    async fn send(&self, req: SendToDeviceRequest) -> Result<()>;
+pub trait EventHandler {
+    fn filters(&self) -> &[Filter];
```

Comment:
```suggestion
pub trait EventChannel {
    fn filters(&self) -> &[Filter];
```

Reply:
(see other comment related to EventHandler)

Reply:
Done.